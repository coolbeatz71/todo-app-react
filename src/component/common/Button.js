import React from 'react';
import PropTypes from 'prop-types';

/**
 * Button component
 * @returns {void}
 * @param {object} props
 */
const Button = ({ children, onClick, type, disabled, myClass }) => (
  <button type={type} disabled={disabled} onClick={onClick} className={`button ${myClass}`}>
    {children}
  </button>
);

Button.propTypes = {
  type: PropTypes.string,
  children: PropTypes.any.isRequired,
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  myClass: PropTypes.string,
};

Button.defaultProps = {
  type: 'button',
  disabled: false,
  myClass: '',
};

export default Button;
