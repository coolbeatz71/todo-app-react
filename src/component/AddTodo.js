import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import Button from './common/Button';
import Input from './common/Input';

/**
 * Add Todo component
 *
 * @class AddTodo
 * @extends {Component}
 */
class AddTodo extends Component {
  state = {
    text: 'Add',
    title: '',
  };

  /**
   * @param {Event} e
   * @memberof AddTodo
   * @returns {void}
   */
  onChange = e => this.setState({ [e.target.name]: e.target.value });

  /**
   * Submit the new todo
   * @param {Event} e
   * @memberof AddTodo
   * @returns {void}
   */
  onClick = e => {
    const { addTodo } = this.props;
    const { title } = this.state;
    e.preventDefault();
    addTodo(title);
    this.setState({ title: '' });
  };

  /**
   * Render the component
   *
   * @returns {void}
   * @memberof AddTodo
   */
  render() {
    const { text, title } = this.state;
    return (
      <div className="row-wrapper">
        <Input value={title} onChange={this.onChange} />
        <Button type="submit" onClick={this.onClick} myClass="btn-primary">
          {text}
        </Button>
      </div>
    );
  }
}

// PropTypes
AddTodo.propTypes = {
  addTodo: PropTypes.func.isRequired,
};

export default AddTodo;
