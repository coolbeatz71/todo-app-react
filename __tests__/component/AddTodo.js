import { shallow, mount } from 'enzyme';
import React from 'react';
import AddTodo from '../../src/component/AddTodo';
import Button from '../../src/component/common/Button';

let wrapper;
const props = {
  addTodo: jest.fn(),
  onClick: jest.fn(),
};
describe('AddTodo component', () => {
  beforeEach(() => {
    wrapper = mount(<AddTodo {...props} />);
  });

  it('Should render the AddTodo component', () => {
    wrapper = shallow(<AddTodo {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should set state `value`', () => {
    const title = 'Title for testing';
    wrapper.setState({
      title,
    });

    wrapper.find('Input').simulate('change', { target: { value: title } });
    expect(wrapper.state('title')).toBe(title);
  });

  it('should trigger Button click`', () => {
    const title = 'Title for testing again';
    wrapper.setState({
      title,
    });

    wrapper.find('Input').simulate('change', { target: { value: title } });
    const btnWrap = shallow(<Button {...props}>Add</Button>);

    btnWrap.find('button').simulate('click');
    expect(wrapper.state('title')).toBe(title);
    expect(props.onClick).toHaveBeenCalled();
  });
});
