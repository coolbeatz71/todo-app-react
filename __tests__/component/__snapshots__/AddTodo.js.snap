// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo component Should render the AddTodo component 1`] = `ShallowWrapper {}`;
